### PR TITLE
[TT-9628] Reset fields inside classic API def but removed from OAS

### PR DESCRIPTION
**Pull request by furkansenharputlu**
<details open>
  <summary><a href="https://tyktech.atlassian.net/browse/TT-9628" title="TT-9628" target="_blank">TT-9628</a></summary>
  <br />
  <table>
    <tr>
      <th>Summary</th>
      <td>Accept CRUD operations of OAS APIs with DB stored API definition</td>
    </tr>
    <tr>
      <th>Type</th>
      <td>
        Story
        Story
      </td>
    </tr>
    <tr>
      <th>Status</th>
      <td>In Code Review</td>
    </tr>
    <tr>
      <th>Points</th>
      <td>N/A</td>
    </tr>
    <tr>
      <th>Labels</th>
      <td><a href="https://tyktech.atlassian.net/issues?jql=project%20%3D%20TT%20AND%20labels%20%3D%20Carnival%20ORDER%20BY%20created%20DESC" title="Carnival">Carnival</a>, <a href="https://tyktech.atlassian.net/issues?jql=project%20%3D%20TT%20AND%20labels%20%3D%20customer_request%20ORDER%20BY%20created%20DESC" title="customer_request">customer_request</a></td>
    </tr>
  </table>
</details>

---

This PR introduces the ability to reset fields in the classic API definition that have been populated when OAS removes corresponding fields.

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: failure** :no_entry_sign: 
Branch used: refs/pull/5419/merge
Commit:  
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5860103451)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5419/merge
Commit: 4b3fd1dfc05a0d898a721fe32c04ea1db969f0fe 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5861511946)

File: apidef/oas/root_test.go [hunk 1]
```diff
@@ -187,6 +187,74 @@ func Fill(t *testing.T, input interface{}, index int) {
 	}
 }
 
+// getNonEmptyFields returns non-empty fields inside a struct.
+func getNonEmptyFields(data interface{}, prefix string) (fields []string) {
```
Comment (titpetric) [inline apidef/oas/root_test.go:191]: This would go into `internal/reflect` for a reuse + unit test needed

**Comment by Tyk-ITS** [comment]
<p>
    Let's make that PR title a 💯 shall we? 💪
    </p>
    <p>
    Your <em>PR title</em> and <em>story title</em> look <strong>slightly different</strong>. Just checking in to know if it was intentional!
    </p>
    <table>
      <tr>
        <th>Story Title</th>
        <td>Accept CRUD operations of OAS APIs with DB stored API definition</td>
      </tr>
      <tr>
          <th>PR Title</th>
          <td>[TT-9628] Reset fields inside classic API def but removed from OAS</td>
        </tr>
    </table>
    <p>
      Check out this <a href="https://www.atlassian.com/blog/git/written-unwritten-guide-pull-requests">guide</a> to learn more about PR best-practices.
    </p>

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5419/merge
Commit: c194d4d4784121002f2491cb33a70bf3bb432876 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5883513644)

File: apidef/oas/oas_test.go [hunk 2]
```diff
@@ -127,6 +127,209 @@ func TestOAS_ExtractTo_DontTouchExistingClassicFields(t *testing.T) {
...
+	vInfo.ExtendedPaths.MockResponse = nil
+	vInfo.ExtendedPaths.Cached = nil
+	vInfo.ExtendedPaths.ValidateJSON = nil
+	a.VersionData.Versions[""] = vInfo
+
+	assert.Empty(t, a.Name)
+
+	noOASSupportFields := getNonEmptyFields(a, "APIDefinition")
+
+	expectedFields := []string{
```
[the remark this answers is not in the document]
Comment (furkansenharputlu) [reply apidef/oas/oas_test.go:178]: I couldn't understand what you mean. This list shows the fields OAS doesn't touch.

File: apidef/oas/root_test.go (hunk 1 above)
Comment (furkansenharputlu) [reply apidef/oas/root_test.go:191]: This one is specific to just `TestOAS_ExtractTo_ResetAPIDefinition`. Very specific. So I believe no need to put it to a common place for reusing.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5419/merge
Commit: 507224aafc88cda545cd2dd4035e92a773ff9258 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/5927637940)

**Review by jeffy-mathew — APPROVED** [review]

**Pull request merged**